src/ts/utils/prop-functions.ts

```diff
@@ -0,0 +1,70 @@
+/**
+ * Utility to allow passing a function as a prop from a dash app.
+ * For example label={'function': 'myLabel'}
+ * where 'myLabel' is a function defined in .js file in /assets
+ *
+ * Based on https://github.com/emilhe/dash-extensions-js
+ */
+
+
+function isPlainObject(o) {
+   return (o === null || o === undefined || Array.isArray(o) || typeof o == 'function' || o.constructor === Date ) ?
+           false
+          :(typeof o == 'object');
+}
+
+function isFunction(functionToCheck) {
+   return functionToCheck && {}.toString.call(functionToCheck) === '[object Function]';
+}
+
+
+export function resolveProp(prop, context = {}) {
+    // If it's not an object, just return.
+    if (!isPlainObject(prop)){
+        return prop
+    }
+    // Check if the prop should be resolved a variable.
+    if (prop.function){
+        return resolveVariable(prop, context)
+    }
+    // Check if the prop should be resolved as an arrow function.
+    if (prop.arrow){
+        return (...args) => prop.arrow
+    }
+    // If none of the special properties are present, do nothing.
+    return prop
+}
+
+function resolveVariable(prop, context){
+    const variable = getDescendantProp(window, prop.function)
+
+    // If it's not there, raise an error.
+    if(variable === undefined){
+        throw new Error("No match for [" + prop.function + "] in the global window object.")
```

Comment:
Nice! Can we say "in window.dashMantineFunctions"?

Reply:
Yes that's the default.  But it also allows you to create other namespaces.  You can use them like this:
{"function": "myNamespce.myFunction"

Reply:
I'd also like to improve the message to include something about syntax errors.  If there are any errors in the namespace then you will see this error message, which is a little confusing.

Reply:
> But it also allows you to create other namespaces.

Oh I missed that - seems a little risky actually to let people access any code on `window`, also a little confusing in that if you want something directly inside `dashMantineFunctions` you can omit the namespace but if you want something nested inside `dashMantineFunctions` you need to include it. I'd be more comfortable if this path always started at `dashMantineFunctions`.